domain/src/test/java/org/oppia/android/domain/feedbackreporting/FeedbackReportManagementControllerTest.kt

```diff
@@ -0,0 +1,445 @@
+      "feedback_reporting_unformatted.json"
+    )
+
+    val request = mockWebServer.takeRequest(
+      timeout = testCoroutineDispatcher.DEFAULT_TIMEOUT_SECONDS,
+      unit = testCoroutineDispatcher.DEFAULT_TIMEOUT_UNIT
+    )
+    request?.let {
+      assertThat(it.method).isEqualTo("POST")
+      // Append newline to the request body to match new line at the end of the JSON file.
+      assertThat(it.body.readUtf8() + "\n").isEqualTo(feedbackReportUnformattedJson)
+    }
+  }
+
+  @Test
+  fun testController_submitFeedbackReport_withoutNetwork_doesNotSendRequestToServer() {
+    mockWebServer.enqueue(MockResponse().setBody("{}"))
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    feedbackReportManagementController.submitFeedbackReport(laterSuggestionReport)
+
+    val request = mockWebServer.takeRequest(
+      timeout = testCoroutineDispatcher.DEFAULT_TIMEOUT_SECONDS,
+      unit = testCoroutineDispatcher.DEFAULT_TIMEOUT_UNIT
+    )
+    assertThat(request).isNull()
+  }
+
+  @Test
+  fun testController_submitFeedbackReport_withNetwork_doesNotSaveReportToStore() {
+    // Enqueue a responses so that the MockWebServer knows when the request is successfully sent.
+    mockWebServer.enqueue(MockResponse().setBody("{}"))
+    networkConnectionUtil.setCurrentConnectionStatus(LOCAL)
+    feedbackReportManagementController.submitFeedbackReport(laterSuggestionReport)
+
+    val reportsStore = feedbackReportManagementController.getFeedbackReportStore().toLiveData()
+    reportsStore.observeForever(this.mockReportsStoreObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+
+    verify(mockReportsStoreObserver, atLeastOnce()).onChanged(reportStoreResultCaptor.capture())
+
+    val reportsList = reportStoreResultCaptor.value
+      .getOrDefault(FeedbackReportingDatabase.getDefaultInstance())
+      .reportsList
+    assertThat(reportsList).isEmpty()
+  }
+
+  @Test
+  fun testController_submitFeedbackReport_withoutNetwork_savesReportToStore() {
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    feedbackReportManagementController.submitFeedbackReport(laterSuggestionReport)
+
+    val reportsStore = feedbackReportManagementController.getFeedbackReportStore().toLiveData()
+    reportsStore.observeForever(mockReportsStoreObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+
+    verify(mockReportsStoreObserver, atLeastOnce()).onChanged(reportStoreResultCaptor.capture())
+
+    val report = reportStoreResultCaptor.value.getOrThrow().getReports(0)
+    assertThat(report.reportSubmissionTimestampSec).isEqualTo(LATER_TIMESTAMP)
+    assertThat(report.userSuppliedInfo.suggestion.suggestionCategory)
+      .isEqualTo(SuggestionCategory.LANGUAGE_SUGGESTION)
+    assertThat(report.userSuppliedInfo.suggestion.userSubmittedSuggestion)
+      .isEqualTo(languageSuggestionText)
+  }
+
+  @Test
+  fun testController_submitMultipleFeedbackReports_withoutNetwork_savesAllReportsToStore() {
```

Comment:
Should we add 1 more test for `submitMultipleFeedbackReports_withNetwork`?

Reply:
Ack -- I initially left this case out since it's the same as `testController_submitFeedbackReport_withNetwork_doesNotSaveReportToStore` multiple times but this might catch something.
Test added.